src/interpreter/arithmetic_operators.rs

```diff
@@ -0,0 +1,182 @@
+use bytecode::instructions::Value;
+use bytecode::instructions::Value::*;
+
+/// Converts the given values using the following rules:
+/// 1. if one of the arguments is a float, then convert both args to Float.
+/// 2. if one of the arguments is a string, then convert both args to Float.
+/// 3. if both arguments are integers, then simply return two Integer values.
+/// # Panics
+/// This panics when one of the arguments is a Bool or a Nil.
+fn to_int_or_float(lhs: &Value, rhs: &Value) -> (Value, Value) {
```

Comment:
This is a real combinatorial explosion. Perhaps `Value` could have a function called `to_int` that returns an `Option<Value>` (`Some(i)` if it could convert it or `None` otherwise)?

Reply:
[Whatever happens, you don't want to do a pairwise conversion: convert one value at a time.]

Reply:
You are right, that makes more sense!